app/Traits/ClearsGlobalSearchCache.php

```
@@ -10,77 +10,119 @@ trait ClearsGlobalSearchCache
+            if (property_exists($this, 'team_id') || isset($this->team_id)) {
+                return $this->team_id ?? null;
+            }
+
+            return null;
+        } catch (\Throwable $e) {
+            // If we can't determine team ID, return null
+            ray('Failed to get team ID for cache: '.$e->getMessage());
+
+            return null;
+        }
     }
```

PHPMD (2.15.0)

The method getTeamIdForCache() has a Cyclomatic Complexity of 10. The configured cyclomatic complexity threshold is 10. (undefined)

(CyclomaticComplexity)